### ISSUE TITLE
Identical Formulas for BatchNorm and LayerNorm

### ISSUE THREAD

**Issue by d-kleine** [pytorch/pytorch#140640]
### 📚 The doc issue

I have noticed that Batch Normalization (BN) and Layer Normalization (LN) are using the same formula in the docs:

- [BatchNorm1d](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html) 
- [BatchNorm2d](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm2d.html)
- [BatchNorm3d](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm3d.html)
- [LayerNorm](https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html)

$$
y = \frac{x - \mathrm{E}[x]}{\sqrt{\mathrm{Var}[x] + \epsilon}} * \gamma + \beta
$$

What is not represented in the BN formula is that mean and variance are computed over the **mini-batch**.

Also, there are no indices specifying how it operates on individual elements of a vector or tensor.

### Suggest a potential alternative/fix

BN (without indices):

$$
y = \frac{x - \mathrm{E}\_{\text{batch}}[x]}{\sqrt{\mathrm{Var}\_{\text{batch}}[x] + \epsilon}} * \gamma + \beta
$$

**Comment by ezyang** [comment]
Sure, send us a patch

**Comment by d-kleine** [comment]
I just noticed that it applies to InstanceNorm too: 

- [InstanceNorm1d](https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm1d.html)
- [InstanceNorm2d](https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm2d.html)
- [InstanceNorm3d](https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm3d.html)

Phew...

**Comment by d-kleine** [comment]
In the docs for these formulas, there are also some really big mistakes, for instance:

```python
class LayerNorm(Module):
    r"""Applies Layer Normalization over a mini-batch of inputs.
...
"""
```

https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html

This describes BatchNorm, not LayerNorm.
